f3/module.go

```diff
@@ -0,0 +1,140 @@
+package f3
+
+import (
+	"context"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+	peer "github.com/libp2p/go-libp2p/core/peer"
+
+	"go.uber.org/multierr"
+	"golang.org/x/xerrors"
+)
+
+type NetworkName string
+
+func (nn NetworkName) DatastorePrefix() datastore.Key {
+	return datastore.NewKey("/f3/" + string(nn))
+}
+func (nn NetworkName) PubSubTopic() string {
+	return "/f3/granite/0.0.1/" + string(nn)
+}
+
+type Module struct {
+	NetworkName NetworkName
+	CertStore   *CertStore
+
+	ds     datastore.Datastore
+	host   host.Host
+	pubsub *pubsub.PubSub
+	verif  Verifier
+	sigs   Signer
+	ec     ECBackend
+
+	topic *pubsub.Topic
+}
+
+// NewModule creates and setups new libp2p f3 module
+// context is used for initialization not runtime
+// TODO notification about new EC chains
+// TODO FCEX
+// TODO flesh out EC notifications
+// TODO flesh out signing backend notifications
```

Comment:
Ditto about moving TODOs into issues. Document what the current code _does_ do, not what it doesn't.